simple/testdata/LintNilCheckAroundRange.go

```diff
@@ -0,0 +1,11 @@
+package pkg
+
+func fn() {
+	str := []string{}
+
+	if str != nil { // MATCH /unnecessary nil check around range/
```

Comment:
Add a couple more test cases, namely:

- a range with no check around it (shouldn't match)
- a check with more statements than just the range (shouldn't match)
- ranging over channels (shouldn't match)
- ranging over maps (shouldn't match)